public/app/plugins/datasource/elasticsearch/configuration/ConfigEditor.tsx

```diff
@@ -43,27 +61,30 @@ export const ConfigEditor = (props: Props) => {
         docsLink="https://grafana.com/docs/grafana/latest/datasources/elasticsearch"
         hasRequiredFields={false}
       />
-
       <Divider />
-
-      <DataSourceHttpSettings
-        defaultUrl="http://localhost:9200"
-        dataSourceConfig={options}
-        showAccessOptions={showAccessOptions.current}
-        onChange={onOptionsChange}
-        sigV4AuthToggleEnabled={config.sigV4AuthEnabled}
-        renderSigV4Editor={<SIGV4ConnectionConfig {...props}></SIGV4ConnectionConfig>}
-        secureSocksDSProxyEnabled={config.secureSocksDSProxyEnabled}
-      />
-
+      <ConnectionSettings config={options} onChange={onOptionsChange} />
+      <Divider />
+      <div className={options.jsonData.sigV4Auth ? styles.sigV4authContainer : ''}>
+        <Auth
+          {...authProps}
+          onAuthMethodSelect={(method) => {
+            options.jsonData.sigV4Auth = method === 'custom-sigv4';
```

Comment:
Instead of modifying `options.jsonData.sigV4Auth` object, could we create a state variable that will hold information if sigV4Auth should be rendered or no and use that instead.

Reply:
Not sure if I fully understand the suggestion, but I'm following the suggestion here for custom methods: https://github.com/grafana/grafana-experimental/tree/main/src/ConfigEditor/Auth#adding-your-own-auth-methods

Reply:
But we need to update `options.jsonData.sigV4Auth` and, because it's a custom authentication method, we need to do that, the component doesn't do it for us.

Reply:
To further ellaborate because I didn't mention this until now, it doesn't work unless I explicitly modify the option:

https://github.com/grafana/grafana/assets/1069378/d5ade0f7-a8c5-43b8-add2-3408d0da5521

Reply:
In any case, I will mention this to the grafana-experimental people because there might be a bug on their side.

Reply:
Gave it a second look and there's no way to use `onOptionsChange()` in combination with `authProps.onAuthMethodSelect()`, because they override each other. I think this approach is not that bad, as it works and it's also related with SigV4 with seems to be niche and could be removed. Let me know what you think.

Reply:
I would solve the issue first and then implement these changes as I don't like that we are mutating options directly, but also then doing it trough `onAuthMethodSelect`. We will need to change it the anyway. I also noticed that Prometheus also implements this, so I would check with them if they have the same issue or how they resolved it.

Reply:
I just checked and it's currently broken for them, the change is not stored.

Reply:
I would then probably create an issue for a team working on a new auth to fix the issue with `onAuthMethodSelect` and custom auth option. Also we should probably mention that we are blocked on this until it is resolved. 

I don't think it make sense to merge this with workaround, as they will need to properly fix it quite soon, as there are probably other data source plugins that support custom methods.

Reply:
Definitely. They are aware and I'll wait until this component has better support for custom auth methods.

Reply:
Fixed by adding a custom onChange implementation that considers SigV4 along the rest of the methods.